src/figuresClasses.ts

```diff
@@ -2,10 +1,98 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+  ) {
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.color = color;
+    this.shape = 'triangle';
+
+    if (
+      this.a === 0 ||
+      this.b === 0 ||
+      this.c === 0 ||
+      this.a >= this.b + this.c ||
+      this.b >= this.a + this.c ||
+      this.c >= this.a + this.b
+    ) {
+      throw new Error('your error message');
```

Comment:
The error thrown here lacks a detailed description. Provide a more descriptive message that explains why the error is being thrown, for example, 'Triangle sides do not form a valid triangle.'